saviSimulation/savi/jason_processing/UaVBehavior.java

```diff
@@ -86,17 +47,39 @@ else if (action.equals("turn(right)"))
 				this.compasAngle += Math.PI/16.0;
 				//Normalize compass angle between 0 and 2 Pi
 				if(compasAngle>2*Math.PI) compasAngle-=2*Math.PI;
-			else if (action.equals( "thrust(on)")) 
+			else if (action.equals("thrust(on)"))
 				this.speedVal = SPEED;
 			else if (action.equals("thrust(off)")) 
 				this.speedVal = 0;
-			else if (action.equals( "thrust(up)")) 
+			else if (action.equals("thrust(up)")) 
 				this.verticalSpeedVal = VERTICAL_SPEED;
 			else if (action.equals("thrust(down)")) 
 				this.verticalSpeedVal = -VERTICAL_SPEED;
 			else if (action.equals("hover")) 
-				this.verticalSpeedVal = 0;	
+				this.verticalSpeedVal = 0;
 				
 		}					
-	}	
+	}
+
+	
+	protected PVector calculateMovementVector (double timeElapsed) {
+		double cosv = Math.cos(this.compasAngle);
+		double sinv = Math.sin(this.compasAngle);
+		PVector movementVector = new PVector((float)(cosv*this.speedVal*timeElapsed), (float)(sinv*this.speedVal*timeElapsed), (float)0.0);
+		movementVector.z+=verticalSpeedVal;
```

Comment:
as noted by @MikeVezina in his PR, verticalSpeed should also depend on time elapsed